modules/client/src/internal/client/encoding.ts

```diff
@@ -53,22 +57,47 @@ export class ClientEncoding extends ClientCore implements IClientEncoding {
   public applyInstallationAction(
     daoAddress: string,
     params: ApplyInstallationParams,
-  ): DaoAction {
+  ): DaoAction[] {
     if (!isAddress(daoAddress)) {
       throw new InvalidAddressError();
     }
+    const provider = this.web3.getProvider();
+    if (!provider) {
+      throw new NoProviderError();
+    }
+    const network = provider.network.name as SupportedNetworks;
+    if (!SupportedNetworksArray.includes(network)) {
+      throw new UnsupportedNetworkError(network);
+    }
```

Comment:
Didn't we have a way to encapsulate this?

Reply:
nope